axum-extra/src/response/file_stream.rs

```diff
@@ -0,0 +1,212 @@
+use axum::{
+    body,
+    response::{IntoResponse, Response},
+    BoxError,
+};
+use bytes::Bytes;
+use futures_util::TryStream;
+use http::{header, StatusCode};
+
+/// Encapsulate the file stream.
+/// The encapsulated file stream construct requires passing in a stream
+/// # Examples
+///
+/// ```
+/// use axum::{
+///     http::StatusCode,
+///     response::{Response, IntoResponse},
+///     Router,
+///     routing::get
+/// };
+/// use axum_extra::response::file_stream::FileStream;
+/// use tokio::fs::File;
+/// use tokio_util::io::ReaderStream ;
+/// async fn file_stream() -> Result<Response, (StatusCode, String)> {
+///     let stream=ReaderStream::new(File::open("test.txt").await.map_err(|e| (StatusCode::NOT_FOUND, format!("File not found: {e}")))?);
+///     let file_stream_resp = FileStream::new(stream)
+///         .file_name("test.txt");
+//
+///     Ok(file_stream_resp.into_response())
+/// }
+/// let app = Router::new().route("/FileStreamDownload", get(file_stream));
+/// # let _: Router = app;
+/// ```
+#[derive(Debug)]
+pub struct FileStream<S>
+where
+    S: TryStream + Send + 'static,
+    S::Ok: Into<Bytes>,
+    S::Error: Into<BoxError>,
+{
+    /// stream.
+    pub stream: S,
+    /// The file name of the file.
+    pub file_name: Option<String>,
+    /// The size of the file.
+    pub content_size: Option<u64>,
+}
+
+impl<S> FileStream<S>
+where
+    S: TryStream + Send + 'static,
+    S::Ok: Into<Bytes>,
+    S::Error: Into<BoxError>,
+{
+    /// Create a file stream.
+    pub fn new(stream: S) -> Self {
+        Self {
+            stream,
+            file_name: None,
+            content_size: None,
+        }
+    }
+
+    /// Set the file name of the file.
```

Comment:
Should we add a `from_file` function where we read the file, set the name and the content size?

Reply:
Filenames and capitalization are not mandatory and the parameter is a stream, so I didn't do it.